src/WebApi/Features/TeamEndpoints.cs

```diff
@@ -39,14 +42,29 @@ public static void MapTeamEndpoints(this WebApplication app)
 
         group
             .MapPost("/{teamId:guid}/heroes/{heroId:guid}",
-                async (ISender sender, Guid teamId, Guid heroId, CancellationToken ct) =>
+                async Task<Results<ValidationProblem, NotFound<string>, Created>> (
+                    ISender sender,
+                    Guid teamId,
+                    Guid heroId,
+                    CancellationToken ct) =>
                 {
                     var command = new AddHeroToTeamCommand(teamId, heroId);
-                    await sender.Send(command, ct);
-                    return Results.Ok();
+                    var result = await sender.Send(command, ct);
+
+                    if (result.IsInvalid())
+                    {
+                        return TypedResultsExt.ValidationProblem(result);
+                    }
+
+                    if (result.IsNotFound())
+                    {
+                        return TypedResultsExt.NotFound(result); // TODO: Add not found details
+                    }
+
+                    return TypedResults.Created();
                 })
             .WithName("AddHeroToTeam")
-            .ProducesPost();
+            .ProducesProblem(StatusCodes.Status500InternalServerError); // TODO: Can we simplify this?
```

Comment:
if your extension returns a ValidationProblem, do you still need to add this decorator?

Reply:
No, because it's inferred base on the function signature:

`async Task<Results<ValidationProblem, NotFound<string>, Created>>`

Reply:
So why not remove the `.ProducesProblem()` decorator?

Reply:
Because that will show up as an error as the minimal API doesn't return a 500 itself.  That's handled by the global exception handler.

<img width="864" alt="image" src="https://github.com/SSWConsulting/SSW.CleanArchitecture/assets/2636640/433163f3-4ae4-4f0a-95f3-75167d3323bf">